### ISSUE TITLE
Add missing metrics

### ISSUE THREAD

**Issue by kakkoyun** [parca-dev/parca-agent#666]
- [ ]  BatchWriteClient (latency, retries)

**Comment by heylongdacoder** [comment]
Hello, is this issue friendly to newcomer? If it is, maybe I can help on this.

**Comment by kakkoyun** [comment]
> Hello, is this issue friendly to newcomer? If it is, maybe I can help on this.

Yes, definitely. Please go ahead.

**Comment by kakkoyun** [comment]
> @kakkoyun regarding the latency, currently from the metrics page, I saw there is a `grpc_client_handling_seconds` histogram. Is this already represent the BatchWriteClient latency?
> 
> Metrics example:
> 
> ```
> grpc_client_handling_seconds_bucket{grpc_method="WriteRaw",grpc_service="parca.profilestore.v1alpha1.ProfileStoreService",grpc_type="unary",le="0.005"} 1
> ```

@heylongdacoder Yes, I get your point. The gRPC client metrics middleware should capture the network latency of the API. The proposed metrics should cover the overall latency with the retries, I think.

**Comment by heylongdacoder** [comment]
got it!
